code/go/0chain.net/smartcontract/dbs/postgresql/db.go

```diff
@@ -0,0 +1,58 @@
+package postgresql
+
+import (
+	"fmt"
+
+	"0chain.net/chaincore/config"
+	"0chain.net/smartcontract/dbs"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type PostgresDB struct {
```

Comment:
we already have script for postgres db

Reply:
I don't follow.

This creates a link to the parent database. So here, `postgres` is the name of a particular PostgreSQL database. 

We can't clone a database with a template; say `event_db` if there are any active connections to the template database. Hence we can't use a connection to a database to clone itself. We have to use another database - hence we open a connection to `postgres` the 'parent' database. 

I put the code in the `postgresql` package as it's a PostgreSQL-specific solution. 

It's the only way I could get the whole thing to work, but welcome other suggestions.